######################Figures 1 and 2., Supplementary Figures 1 and 5.#####################
.libPaths(c(.libPaths(), "/sc/wo/home/karmajx/R/x86_64-pc-linux-gnu-library/3.4", "/sc/wo/home/karmajx/R/x86_64-pc-linux-gnu-library/3.5"))
options(stringsAsFactors = F)
setwd('/sc/wo/tri_data/jk/2018_IPF_publicdata_JK/')
sapply(c('rcompanion', 'ggplot2', 'genefilter', 'WGCNA', 'reshape2', 'pheatmap', 'RColorBrewer', 'limma', 'edgeR', 'cowplot',
         'ConsensusClusterPlus', 'xlsx', 'GSVA', 'clusterProfiler', 'DOSE', 'GO.db', 'org.Hs.eg.db', 'GSEABase', 'Pigengene',
         'ReactomePA', 'magrittr', 'FactoMineR', 'factoextra', 'FSA', 'circlize', 'tibble', 'dplyr', 'corrplot', 'ggcorrplot'), library, character.only=T)

#consensus clustering, PCA and limma
gse47460_pheno <- read.csv('gse47460_pheno_clean.csv', header = T, row.names = 1) # contains all samples in GSE47460
gse47460_matrix <- read.csv('gse47460_as_matrix_platformremoved.csv', header = T, row.names = 1) 
# matrix generated by reading raw data from GEO by ArrayStudio, followed by quantile normalization and batch effect removal (see steps below)
# GSE47460 features two platforms, GPL6480 and GPL14550
# data from two platforms was processed separately from author-provided raw data using ArrayStudio
# using normalized data, probes were collapsed to genes for both sample sets using median of probes
# data from two platforms was merged based on common set of genes
# batch effect of platform removed by ArrayStudio 'Remove batch effect' feature, resulting in matrix above

set.seed(1)
nicecolors <- c("#999999", "#E69F00", "#56B4E9")
gsm <- vector()
for (i in 1:ncol(gse47460_matrix)){
  gsm <- c(gsm, strsplit(colnames(gse47460_matrix)[i], split = '_', fixed = T)[[1]][1])
}
colnames(gse47460_matrix) <- gsm
gse47460_phenom <- as.data.frame(merge(gse47460_pheno, t(gse47460_matrix), by.x = 0, by.y = 0))
rownames(gse47460_phenom) <- gse47460_phenom$Row.names
gse47460_phenom$Row.names <- NULL
set.seed(1)
d_gse47460 <- gse47460_matrix[,row.names(subset(gse47460_pheno, subtype=='2-UIP/IPF'))] # n=160 samples
mads_gse47460=apply(d_gse47460,1,mad)
d_gse47460=d_gse47460[rev(order(mads_gse47460))[1:5000],]
d_gse47460 = as.matrix(sweep(d_gse47460,1, apply(d_gse47460,1,median,na.rm=T)))
dir.create('gse47460_consensus_081420')
title='gse47460_consensus_081420'
#Figure 1A
results_gse47460 = ConsensusClusterPlus(d_gse47460,maxK=6,reps=1000,pItem=0.8,pFeature=1,clusterAlg="hc",
                                        distance="pearson",seed=1234) # Figure 1A

resultsicl <- calcICL(results_gse47460)
consensusmatrices <- lapply(2:6, function(k) results_gse47460[[k]]$consensusMatrix)
names(consensusmatrices) <- paste0('k', seq(2, 6))
pac <- lapply(consensusmatrices, function(k) diceR::PAC(k, lower = 0.1, upper = 0.9))
pacframe <- as.data.frame(unlist(pac))
colnames(pacframe) <- 'PAC'
pacframe$number_of_clusters <- seq(2, 6)
#Supplementary Figure 1A.
ggplot(pacframe, aes(x=number_of_clusters, y=PAC)) + 
  geom_point(shape=18, color="blue", size = 5)+
  geom_line() + scale_y_continuous(limits = c(0.25, 0.35))

consensusdf_gse47460 <- as.data.frame(results_gse47460[[2]]$consensusClass)
colnames(consensusdf_gse47460) <- 'consensusclass'
consensusdf_gse47460$consensusclass <- paste0('Cluster_', consensusdf_gse47460$consensusclass)
healthydf_gse47460 <- subset(gse47460_pheno, Phenotype=='Control')
healthydf2_gse47460 <- healthydf_gse47460[,1]
names(healthydf2_gse47460) <- row.names(healthydf_gse47460)
healthydf2_gse47460 <- as.data.frame(healthydf2_gse47460)
colnames(healthydf2_gse47460) <- 'consensusclass'
consensusdf2_gse47460 <- rbind(consensusdf_gse47460, healthydf2_gse47460)
#consensusdf2_gse47460$geo <- as.data.frame(unlist(strsplit(row.names(consensusdf2_gse47460), split = '_', fixed = T)))[seq(1, 804, by=3),]
gse47460_phenom_consensus <- as.data.frame(merge(consensusdf2_gse47460, gse47460_phenom, by.x = 0, by.y = 0))
rownames(gse47460_phenom_consensus) <- gse47460_phenom_consensus$Row.names
gse47460_phenom_consensus$Row.names <- NULL

#heatmap in Figure 1B generated using ArrayStudio (Qiagen). Code for this we cannot post here

#Figure 1C:
gse47460_pca <- PCA(gse47460_phenom_consensus[,c(colnames(gse47460_phenom_consensus)[1:18], row.names(d_gse47460))], quali.sup = 1:18, graph = FALSE)
fviz_pca_ind(gse47460_pca, habillage = "consensusclass", addEllipses = TRUE, geom = 'point') # Figure 1C

#Differential expression by limma
gse47460_phenom_consensus2 <- gse47460_phenom_consensus[which(gse47460_phenom_consensus$smoker!='no_value'),] # narrowed analysis to patients with smoking information available so it can be included in linear model
smokervalues <- row.names(subset(gse47460_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2")))
fl_gse47460 <- as.factor(subset(gse47460_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$consensusclass)
sm_gse47460 <- as.factor(subset(gse47460_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$smoker)
g_gse47460 <- as.factor(subset(gse47460_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$Sex)
design_gse47460 <- model.matrix(~0+fl_gse47460+sm_gse47460+g_gse47460)
gse47460_matrix2<- gse47460_matrix[,row.names(subset(gse47460_phenom_consensus2, gse47460_phenom_consensus2$Phenotype=='Interstitial lung disease'))]
fit_gse47460 <- lmFit(gse47460_matrix2, design_gse47460)
cont.matrix_gse47460 <- makeContrasts(fl_gse47460Cluster_1-fl_gse47460Cluster_2, levels=design_gse47460)
fit2_gse47460 <- contrasts.fit(fit_gse47460, cont.matrix_gse47460)
fit2_gse47460 <- eBayes(fit2_gse47460)
tT_gse47460 <- topTable(fit2_gse47460, adjust="fdr", sort.by="B", number=nrow(gse47460_matrix))
tT_gse47460$gene <- row.names(tT_gse47460)
tT_gse47460_sig <- subset(tT_gse47460, adj.P.Val<0.05 & abs(logFC)>0.58)

fl_gse47460_vshealthy <- as.factor(gse47460_phenom_consensus2$consensusclass)
sm_gse47460_vshealthy <- as.factor(gse47460_phenom_consensus2$smoker)
g_gse47460_vshealthy <- as.factor(gse47460_phenom_consensus2$Sex)

design_gse47460_vshealthy <- model.matrix(~0+fl_gse47460_vshealthy+sm_gse47460_vshealthy+g_gse47460_vshealthy)
fit_gse47460_vshealthy <- lmFit(t(gse47460_phenom_consensus2[,15:19010]), design_gse47460_vshealthy)
cont.matrix_gse47460_vshealthy <- makeContrasts(fl_gse47460_vshealthyCluster_1-fl_gse47460_vshealthyControl, fl_gse47460_vshealthyCluster_2-fl_gse47460_vshealthyControl,
                                                levels=design_gse47460_vshealthy)
cont.matrix_gse47460_1vshealthy <- makeContrasts(fl_gse47460_vshealthyCluster_1-fl_gse47460_vshealthyControl, levels=design_gse47460_vshealthy)
cont.matrix_gse47460_2vshealthy <- makeContrasts(fl_gse47460_vshealthyCluster_2-fl_gse47460_vshealthyControl, levels=design_gse47460_vshealthy)

fit2_gse47460_vshealthy <- contrasts.fit(fit_gse47460_vshealthy, cont.matrix_gse47460_vshealthy)
fit2_gse47460_vshealthy <- eBayes(fit2_gse47460_vshealthy)
tT_gse47460_vshealthy <- topTable(fit2_gse47460_vshealthy, adjust="fdr", sort.by="B", number=nrow(gse47460_matrix))
tT_gse47460_vshealthy$gene <- row.names(tT_gse47460_vshealthy)
colnames(tT_gse47460_vshealthy)[1:2] <- c('subset1', 'subset2')
tT_gse47460_vshealthy_sig1 <- subset(tT_gse47460_vshealthy, abs(tT_gse47460_vshealthy$subset1)>0.58 & tT_gse47460_vshealthy$adj.P.Val<0.05)
tT_gse47460_vshealthy_sig2 <- subset(tT_gse47460_vshealthy, abs(tT_gse47460_vshealthy$subset2)>0.58 & tT_gse47460_vshealthy$adj.P.Val<0.05)

fit2_gse47460_1vshealthy <- contrasts.fit(fit_gse47460_vshealthy, cont.matrix_gse47460_1vshealthy)
fit2_gse47460_1vshealthy <- eBayes(fit2_gse47460_1vshealthy)
tT_gse47460_1vshealthy <- topTable(fit2_gse47460_1vshealthy, adjust="fdr", sort.by="B", number=nrow(gse47460_matrix))
tT_gse47460_1vshealthy$gene <- row.names(tT_gse47460_1vshealthy)
#colnames(tT_gse47460_1vshealthy)[1:2] <- c('subset1', 'subset2')
tT_gse47460_1vshealthy_sig1 <- subset(tT_gse47460_1vshealthy, abs(tT_gse47460_1vshealthy$logFC)>0.58 & tT_gse47460_1vshealthy$adj.P.Val<0.05)
tT_gse47460_1vshealthy_sig1 <- tT_gse47460_1vshealthy_sig1[order(tT_gse47460_1vshealthy_sig1$logFC, decreasing = T),]
write.xlsx(tT_gse47460_1vshealthy_sig1, file = 'tT_gse47460_1vshealthy_sig1_081720.xlsx')

fit2_gse47460_2vshealthy <- contrasts.fit(fit_gse47460_vshealthy, cont.matrix_gse47460_2vshealthy)
fit2_gse47460_2vshealthy <- eBayes(fit2_gse47460_2vshealthy)
tT_gse47460_2vshealthy <- topTable(fit2_gse47460_2vshealthy, adjust="fdr", sort.by="B", number=nrow(gse47460_matrix))
tT_gse47460_2vshealthy$gene <- row.names(tT_gse47460_2vshealthy)
#colnames(tT_gse47460_2vshealthy)[1:2] <- c('subset1', 'subset2')
tT_gse47460_2vshealthy_sig1 <- subset(tT_gse47460_2vshealthy, abs(tT_gse47460_2vshealthy$logFC)>0.58 & tT_gse47460_2vshealthy$adj.P.Val<0.05)
tT_gse47460_2vshealthy_sig1 <- tT_gse47460_2vshealthy_sig1[order(tT_gse47460_2vshealthy_sig1$logFC, decreasing = T),]
write.xlsx(tT_gse47460_2vshealthy_sig1, file = 'tT_gse47460_2vshealthy_sig1_081720.xlsx')

tT_gse47460_1vshealthy_sig_2vshealthy_sig <- merge(tT_gse47460_1vshealthy_sig1, tT_gse47460_2vshealthy_sig1, by = 'gene')
cor.test(tT_gse47460_1vshealthy_sig_2vshealthy_sig$logFC.x, tT_gse47460_1vshealthy_sig_2vshealthy_sig$logFC.y)

#####################GSE134692 for Figure 1F-J.#######################
#used gene expression matrix and design files posted on NCBI Gene Expression Omnibus
gse134692 <- read.table('GSE134692_tmm_normalized_and_filtered_log2CPM.txt', header = T, stringsAsFactors = F, row.names = 1)
gse134692_annotation <- read.delim('GSE134692_gene_annotation.txt', header = T, sep = '\t', row.names = 1)
gse134692_pheno <- read.delim('GSE134692_design.txt', header = T, row.names = 1, sep = '\t')
table(gse134692_pheno$DiseaseStatus)

set.seed(1)
d_gse134692 <- gse134692[,row.names(subset(gse134692_pheno, gse134692_pheno$DiseaseStatus=='IPF' & gse134692_pheno$Batch==1))]
mads_gse134692=apply(d_gse134692,1,mad)
d_gse134692=d_gse134692[rev(order(mads_gse134692))[1:5000],]
d_gse134692 = as.matrix(sweep(d_gse134692,1, apply(d_gse134692,1,median,na.rm=T)))
dir.create('gse134692_consensus_011921batch1_5000')
title='gse134692_consensus_011921batch1_5000'
#Figure 1F:
results_gse134692 = ConsensusClusterPlus(d_gse134692,maxK=6,reps=1000,pItem=0.8,pFeature=1,clusterAlg="hc", title = title,
                                         distance="pearson",seed=1234, plot = 'png')

resultsicl <- calcICL(results_gse134692)
consensusmatrices <- lapply(2:6, function(k) results_gse134692[[k]]$consensusMatrix)
names(consensusmatrices) <- paste0('k', seq(2, 6))
pac <- lapply(consensusmatrices, function(k) diceR::PAC(k, lower = 0.1, upper = 0.9))
pacframe <- as.data.frame(unlist(pac))
colnames(pacframe) <- 'PAC'
pacframe$number_of_clusters <- seq(2, 6)
#Supplementary Figure 1B.
ggplot(pacframe, aes(x=number_of_clusters, y=PAC)) + 
  geom_point(shape=18, color="blue", size = 5)+
  geom_line()# + scale_y_continuous(limits = c(0.25, 0.35))

consensusdf_gse134692 <- as.data.frame(results_gse134692[[2]]$consensusClass)
colnames(consensusdf_gse134692) <- 'consensusclass'
consensusdf_gse134692$consensusclass <- paste0('Cluster_', consensusdf_gse134692$consensusclass)
consensusdf_gse134692$reverseclass <- plyr::mapvalues(consensusdf_gse134692$consensusclass, 
                                                      from = c('Cluster_1', 'Cluster_2'), to = c('Cluster_2', 'Cluster_1')) # so colors match with other figures
healthydf_gse134692 <- subset(gse134692_pheno, gse134692_pheno$DiseaseStatus=='Normal' & gse134692_pheno$Batch==1)
healthydf2_gse134692 <- healthydf_gse134692[,2]
names(healthydf2_gse134692) <- row.names(healthydf_gse134692)
healthydf2_gse134692 <- as.data.frame(healthydf2_gse134692)
colnames(healthydf2_gse134692) <- 'consensusclass'
healthydf2_gse134692$reverseclass <- rep('Normal', nrow(healthydf2_gse134692))
consensusdf2_gse134692 <- rbind(consensusdf_gse134692, healthydf2_gse134692)
write.csv(consensusdf2_gse134692, file = 'gse134692_batch1_consensusclasses_011921.csv')

gse134692_phenom <- as.data.frame(merge(gse134692_pheno, t(gse134692), by.x = 0, by.y = 0))
rownames(gse134692_phenom) <- gse134692_phenom$Row.names
gse134692_phenom$Row.names <- NULL

gse134692_phenom_consensus <- as.data.frame(merge(consensusdf2_gse134692, gse134692_phenom, by.x = 0, by.y = 0))
rownames(gse134692_phenom_consensus) <- gse134692_phenom_consensus$Row.names
gse134692_phenom_consensus$Row.names <- NULL

#Figure 1G generated in ArrayStudio (Qiagen). We cannot post code for this figure here.
#Figure 1H:
gse134692_pca <- PCA(gse134692_phenom_consensus[,c(colnames(gse134692_phenom_consensus)[1:16], row.names(d_gse134692))], quali.sup = 1:16, graph = FALSE)
fviz_pca_ind(gse134692_pca, habillage = "consensusclass", addEllipses = TRUE, geom = 'point')

#Figure 1I below:
genes <- c('RPGRIP1', 'DNAH6', 'DNAH7', 'DNAI1', 'MUC5B')
library(annotate)
library(org.Hs.eg.db)
humangenes <- useMart('ENSEMBL_MART_ENSEMBL', 'hsapiens_gene_ensembl')
genes_ensembl <- getBM(attributes = c("ensembl_gene_id", "hgnc_symbol") , filters= "hgnc_symbol", values = genes, mart = humangenes)
rm(list=ls(pattern = 'Figure1_'))
rm(figure1_plots)
for (i in genes_ensembl$ensembl_gene_id){
  assign(paste0('Figure1_', i), ggplot(gse134692_phenom_consensus, aes_string('consensusclass', i, fill = 'consensusclass')) + 
           geom_boxplot(width=0.5, show.legend = F) + geom_jitter(color="black", shape=16, position=position_jitter(0.2)) + theme_bw() + 
           scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) + theme(legend.position = "none"))
}
figure1_plots <- mget(ls(pattern = 'Figure1_'))
rm(list=ls(pattern = 'Figure1_'))
do.call(plot_grid, figure1_plots)

fl_gse134692 <- as.factor(subset(gse134692_phenom_consensus, consensusclass %in% c("Cluster_1", "Cluster_2"))$consensusclass)
gse134692_phenom_consensus$tobacco <- gsub(" ", "_", gse134692_phenom_consensus$tobacco, fixed = T) #change spaces to underscores
sm_gse134692 <- as.factor(subset(gse134692_phenom_consensus, consensusclass %in% c("Cluster_1", "Cluster_2"))$tobacco)
g_gse134692 <- as.factor(subset(gse134692_phenom_consensus, consensusclass %in% c("Cluster_1", "Cluster_2"))$gender)
design_gse134692 <- model.matrix(~0+fl_gse134692+sm_gse134692+g_gse134692)
gse134692_matrix2<- gse134692[,row.names(subset(gse134692_phenom_consensus, gse134692_phenom_consensus$DiseaseStatus=='IPF'))]
fit_gse134692 <- lmFit(gse134692_matrix2, design_gse134692)
cont.matrix_gse134692 <- makeContrasts(fl_gse134692Cluster_1-fl_gse134692Cluster_2, levels=design_gse134692)
fit2_gse134692 <- contrasts.fit(fit_gse134692, cont.matrix_gse134692)
fit2_gse134692 <- eBayes(fit2_gse134692)
tT_gse134692 <- topTable(fit2_gse134692, adjust="fdr", sort.by="B", number=nrow(gse134692_matrix2))
tT_gse134692$gene <- row.names(tT_gse134692)
tT_gse134692_sig <- subset(tT_gse134692, adj.P.Val<0.05 & abs(logFC)>0.58)
library(biomaRt)
library(annotate)
library(org.Hs.eg.db)
humangenes <- useMart('ENSEMBL_MART_ENSEMBL', 'hsapiens_gene_ensembl')
gse134692_sig_symbols <- getBM(attributes = c("ensembl_gene_id", "hgnc_symbol") , filters= "ensembl_gene_id", values = row.names(tT_gse134692_sig), mart = humangenes)
gse134692_sig_annot <- merge(tT_gse134692_sig, gse134692_sig_symbols, by.x=0, by.y="ensembl_gene_id")
rownames(gse134692_sig_annot) <- gse134692_sig_annot$Row.names
gse134692_sig_annot$Row.names <- NULL

tT_gse47460_tT_gse134692 <- merge(tT_gse47460_sig, gse134692_sig_annot, by.x = 'gene', by.y = 'hgnc_symbol')
rownames(tT_gse47460_tT_gse134692) <- tT_gse47460_tT_gse134692$gene
colnames(tT_gse47460_tT_gse134692)[c(2,8)] <- c('logFC_GSE47460', 'logFC_GSE134692')

#Figure 1J below:
ggplot(tT_gse47460_tT_gse134692, aes(x=logFC_GSE47460, y=logFC_GSE134692)) + 
  geom_point(shape=18, color="blue")+
  geom_smooth(method=lm,  linetype="dashed",
              color="darkred", fill="blue")
cor.test(tT_gse47460_tT_gse134692$logFC_GSE47460, tT_gse47460_tT_gse134692$logFC_GSE134692)

########################Supplementary Figure 1C-D.####################
gse47460_patient <- read.csv('gse47460_fullpheno.csv', header = T)
gse32537_fullpheno <- read.csv('gse32537_fullpheno.csv', header = T)

gse47460_fullpheno <- merge(gse47460_patient, gse47460_pheno, by.x = 'Geo', by.y = 0)
rownames(gse47460_fullpheno) <- gse47460_fullpheno$Geo

gse47460_fullpheno_IPF <- subset(gse47460_fullpheno, gse47460_fullpheno$subtype=='2-UIP/IPF')
strsplit(gse47460_fullpheno_IPF$Patient[1], split = '^[0-9]')
gse47460_fullpheno_IPF$cleanpatient <- gsub('RL_ILD', '', gse47460_fullpheno_IPF$Patient, fixed = T)
gse47460_fullpheno_IPF$cleanpatient <- gsub('LL_ILD', '', gse47460_fullpheno_IPF$cleanpatient, fixed = T)
gse47460_fullpheno_IPF$cleanpatient <- gsub('LU_ILD', '', gse47460_fullpheno_IPF$cleanpatient, fixed = T)
gse47460_fullpheno_IPF$cleanpatient <- gsub('RU_ILD', '', gse47460_fullpheno_IPF$cleanpatient, fixed = T)
gse47460_fullpheno_IPF$cleanpatient <- gsub('LI_ILD', '', gse47460_fullpheno_IPF$cleanpatient, fixed = T)
gse47460_fullpheno_IPF$cleanpatient <- gsub('RM_ILD', '', gse47460_fullpheno_IPF$cleanpatient, fixed = T)

gse32537_fullpheno_ipf <- subset(gse32537_fullpheno, gse32537_fullpheno$Diagnosis=='IPF/UIP')

length(intersect(gse32537_fullpheno_ipf$Patient, gse47460_fullpheno_IPF$cleanpatient))

gse47460_notin_gse32537 <- subset(gse47460_fullpheno_IPF, gse47460_fullpheno_IPF$cleanpatient %in% setdiff(gse47460_fullpheno_IPF$cleanpatient, gse32537_fullpheno_ipf$Patient))

set.seed(1)
d_gse47460_notin_gse32537 <- gse47460_matrix[,gse47460_notin_gse32537$Geo] # n=75 samples
mads_gse47460_notin_gse32537=apply(d_gse47460_notin_gse32537,1,mad)
d_gse47460_notin_gse32537=d_gse47460_notin_gse32537[rev(order(mads_gse47460_notin_gse32537))[1:5000],]
d_gse47460_notin_gse32537 = as.matrix(sweep(d_gse47460_notin_gse32537,1, apply(d_gse47460_notin_gse32537,1,median,na.rm=T)))
dir.create('gse47460_notin_gse32537_consensus_012921')
title='gse47460_notin_gse32537_consensus_012921'
#Figure 1A
results_gse47460_notin_gse32537 = ConsensusClusterPlus(d_gse47460_notin_gse32537,maxK=6,reps=1000,pItem=0.8,pFeature=1,clusterAlg="hc",
                                                       distance="pearson",seed=1234) # Figure 1A

gse47460_yesin_gse32537 <- subset(gse47460_fullpheno_IPF, gse47460_fullpheno_IPF$cleanpatient %in% intersect(gse47460_fullpheno_IPF$cleanpatient, gse32537_fullpheno_ipf$Patient))

set.seed(1)
d_gse47460_yesin_gse32537 <- gse47460_matrix[,gse47460_yesin_gse32537$Geo] # n=85 samples
mads_gse47460_yesin_gse32537=apply(d_gse47460_yesin_gse32537,1,mad)
d_gse47460_yesin_gse32537=d_gse47460_yesin_gse32537[rev(order(mads_gse47460_yesin_gse32537))[1:5000],]
d_gse47460_yesin_gse32537 = as.matrix(sweep(d_gse47460_yesin_gse32537,1, apply(d_gse47460_yesin_gse32537,1,median,na.rm=T)))
dir.create('gse47460_yesin_gse32537_consensus_012921')
title='gse47460_yesin_gse32537_consensus_012921'

results_gse47460_yesin_gse32537 = ConsensusClusterPlus(d_gse47460_yesin_gse32537,maxK=6,reps=1000,pItem=0.8,pFeature=1,clusterAlg="hc",
                                                       distance="pearson",seed=1234)
ingse32537 <- vector()
for (i in 1:nrow(gse47460_fullpheno_IPF)){
  if (gse47460_fullpheno_IPF$cleanpatient[i] %in% gse32537_fullpheno_ipf$Patient){
    ingse32537[i] <- 'yes'
  }else{
    ingse32537[i] <- 'no'
  }
}
table(ingse32537)
gse47460_fullpheno_IPF$overlap <- ingse32537

gse47460_fullpheno_IPF_consensus <- merge(gse47460_fullpheno_IPF, consensusdf_gse47460, by.x = 'Geo', by.y = 0)
#Supplementary Figure 1C.
table(gse47460_fullpheno_IPF_consensus$consensusclass, gse47460_fullpheno_IPF_consensus$overlap)
fisher.test(table(gse47460_fullpheno_IPF_consensus$overlap, gse47460_fullpheno_IPF_consensus$consensusclass))

############DEGs##############
consensusdf_gse47460_notin_gse32537 <- as.data.frame(results_gse47460_notin_gse32537[[2]]$consensusClass)
colnames(consensusdf_gse47460_notin_gse32537) <- 'consensusclass'
consensusdf_gse47460_notin_gse32537$consensusclass <- paste0('Cluster_', consensusdf_gse47460_notin_gse32537$consensusclass)

healthydf_gse47460_notin_gse32537 <- subset(gse47460_pheno, Phenotype=='Control')
healthydf2_gse47460_notin_gse32537 <- healthydf_gse47460_notin_gse32537[,1]
names(healthydf2_gse47460_notin_gse32537) <- row.names(healthydf_gse47460_notin_gse32537)
healthydf2_gse47460_notin_gse32537 <- as.data.frame(healthydf2_gse47460_notin_gse32537)
colnames(healthydf2_gse47460_notin_gse32537) <- 'consensusclass'
consensusdf2_gse47460_notin_gse32537 <- rbind(consensusdf_gse47460_notin_gse32537, healthydf2_gse47460_notin_gse32537)
gse47460_notin_gse32537_phenom_consensus <- as.data.frame(merge(consensusdf2_gse47460_notin_gse32537, gse47460_phenom, by.x = 0, by.y = 0))
rownames(gse47460_notin_gse32537_phenom_consensus) <- gse47460_notin_gse32537_phenom_consensus$Row.names
gse47460_notin_gse32537_phenom_consensus$Row.names <- NULL

#Differntial expression by limma
gse47460_notin_gse32537_phenom_consensus2 <- gse47460_notin_gse32537_phenom_consensus[which(gse47460_notin_gse32537_phenom_consensus$smoker!='no_value'),] # narrowed analysis to patients with smoking information available so it can be included in linear model
smokervalues <- row.names(subset(gse47460_notin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2")))
fl_gse47460_notin_gse32537 <- as.factor(subset(gse47460_notin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$consensusclass)
sm_gse47460_notin_gse32537 <- as.factor(subset(gse47460_notin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$smoker)
g_gse47460_notin_gse32537 <- as.factor(subset(gse47460_notin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$Sex)
design_gse47460_notin_gse32537 <- model.matrix(~0+fl_gse47460_notin_gse32537+sm_gse47460_notin_gse32537+g_gse47460_notin_gse32537)
gse47460_notin_gse32537_matrix2<- gse47460_matrix[,row.names(subset(gse47460_notin_gse32537_phenom_consensus2, gse47460_notin_gse32537_phenom_consensus2$Phenotype=='Interstitial lung disease'))]
fit_gse47460_notin_gse32537 <- lmFit(gse47460_notin_gse32537_matrix2, design_gse47460_notin_gse32537)
cont.matrix_gse47460_notin_gse32537 <- makeContrasts(fl_gse47460_notin_gse32537Cluster_1-fl_gse47460_notin_gse32537Cluster_2, levels=design_gse47460_notin_gse32537)
fit2_gse47460_notin_gse32537 <- contrasts.fit(fit_gse47460_notin_gse32537, cont.matrix_gse47460_notin_gse32537)
fit2_gse47460_notin_gse32537 <- eBayes(fit2_gse47460_notin_gse32537)
tT_gse47460_notin_gse32537 <- topTable(fit2_gse47460_notin_gse32537, adjust="fdr", sort.by="B", number=nrow(gse47460_matrix))
tT_gse47460_notin_gse32537$gene <- row.names(tT_gse47460_notin_gse32537)
tT_gse47460_notin_gse32537_sig <- subset(tT_gse47460_notin_gse32537, adj.P.Val<0.05 & abs(logFC)>0.58)

consensusdf_gse47460_yesin_gse32537 <- as.data.frame(results_gse47460_yesin_gse32537[[2]]$consensusClass)
colnames(consensusdf_gse47460_yesin_gse32537) <- 'consensusclass'
consensusdf_gse47460_yesin_gse32537$consensusclass <- paste0('Cluster_', consensusdf_gse47460_yesin_gse32537$consensusclass)

healthydf_gse47460_yesin_gse32537 <- subset(gse47460_pheno, Phenotype=='Control')
healthydf2_gse47460_yesin_gse32537 <- healthydf_gse47460_yesin_gse32537[,1]
names(healthydf2_gse47460_yesin_gse32537) <- row.names(healthydf_gse47460_yesin_gse32537)
healthydf2_gse47460_yesin_gse32537 <- as.data.frame(healthydf2_gse47460_yesin_gse32537)
colnames(healthydf2_gse47460_yesin_gse32537) <- 'consensusclass'
consensusdf2_gse47460_yesin_gse32537 <- rbind(consensusdf_gse47460_yesin_gse32537, healthydf2_gse47460_yesin_gse32537)
#consensusdf2_gse47460_yesin_gse32537$geo <- as.data.frame(unlist(strsplit(row.names(consensusdf2_gse47460_yesin_gse32537), split = '_', fixed = T)))[seq(1, 804, by=3),]
gse47460_yesin_gse32537_phenom_consensus <- as.data.frame(merge(consensusdf2_gse47460_yesin_gse32537, gse47460_phenom, by.x = 0, by.y = 0))
rownames(gse47460_yesin_gse32537_phenom_consensus) <- gse47460_yesin_gse32537_phenom_consensus$Row.names
gse47460_yesin_gse32537_phenom_consensus$Row.names <- NULL

#Differential expression by limma
gse47460_yesin_gse32537_phenom_consensus2 <- gse47460_yesin_gse32537_phenom_consensus[which(gse47460_yesin_gse32537_phenom_consensus$smoker!='no_value'),] # narrowed analysis to patients with smoking information available so it can be included in linear model
smokervalues <- row.names(subset(gse47460_yesin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2")))
fl_gse47460_yesin_gse32537 <- as.factor(subset(gse47460_yesin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$consensusclass)
sm_gse47460_yesin_gse32537 <- as.factor(subset(gse47460_yesin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$smoker)
g_gse47460_yesin_gse32537 <- as.factor(subset(gse47460_yesin_gse32537_phenom_consensus2, consensusclass %in% c("Cluster_1", "Cluster_2"))$Sex)
design_gse47460_yesin_gse32537 <- model.matrix(~0+fl_gse47460_yesin_gse32537+sm_gse47460_yesin_gse32537+g_gse47460_yesin_gse32537)
gse47460_yesin_gse32537_matrix2<- gse47460_matrix[,row.names(subset(gse47460_yesin_gse32537_phenom_consensus2, gse47460_yesin_gse32537_phenom_consensus2$Phenotype=='Interstitial lung disease'))]
fit_gse47460_yesin_gse32537 <- lmFit(gse47460_yesin_gse32537_matrix2, design_gse47460_yesin_gse32537)
cont.matrix_gse47460_yesin_gse32537 <- makeContrasts(fl_gse47460_yesin_gse32537Cluster_1-fl_gse47460_yesin_gse32537Cluster_2, levels=design_gse47460_yesin_gse32537)
fit2_gse47460_yesin_gse32537 <- contrasts.fit(fit_gse47460_yesin_gse32537, cont.matrix_gse47460_yesin_gse32537)
fit2_gse47460_yesin_gse32537 <- eBayes(fit2_gse47460_yesin_gse32537)
tT_gse47460_yesin_gse32537 <- topTable(fit2_gse47460_yesin_gse32537, adjust="fdr", sort.by="B", number=nrow(gse47460_matrix))
tT_gse47460_yesin_gse32537$gene <- row.names(tT_gse47460_yesin_gse32537)
tT_gse47460_yesin_gse32537_sig <- subset(tT_gse47460_yesin_gse32537, adj.P.Val<0.05 & abs(logFC)>0.58)

gse47460_yesin_notin_combined <- merge(tT_gse47460_yesin_gse32537_sig, tT_gse47460_notin_gse32537_sig, by = 'gene')
rownames(gse47460_yesin_notin_combined) <- gse47460_yesin_notin_combined$gene
colnames(gse47460_yesin_notin_combined)[c(2,8)] <- c('logFC_overlapping_with_GSE32537', 'logFC_notoverlapping_with_GSE32537')

#Supplementary Figure 1D:
ggplot(gse47460_yesin_notin_combined, aes(x=logFC_overlapping_with_GSE32537, y=logFC_notoverlapping_with_GSE32537)) + 
  geom_point(shape=18, color="blue")+
  geom_smooth(method=lm,  linetype="dashed",
              color="darkred", fill="blue")
cor.test(gse47460_yesin_notin_combined$logFC_overlapping_with_GSE32537, gse47460_yesin_notin_combined$logFC_notoverlapping_with_GSE32537)